test/syscalls/linux/proc.cc

```diff
@@ -511,6 +513,84 @@ TEST(ProcSelfAuxv, EntryValues) {
   EXPECT_EQ(i, proc_auxv.size());
 }
 
```

Comment:
If you don't mind, could you please add more test coverage? /proc/[pid]/mem is quite sensitive since it exposes a process' entire memory layout, so we should test it very thoroughly. At least the following cases came to mind:
- Using an open fd on /proc/[pid]/mem after process [pid] has exited
- Reading /proc/[pid]/mem from a process with different uid/gid combinations
- Reading an unallocated memory region
- Reading past an allocated memory region
- Reading multiple times in sequence (offset should be updated appropriately)
- Seeking

Reply:
So in total I have tests for

- Basic read via PRead
- Read from unmapped region expecting EIO
- Repeated reads
- Repeated seeks, moving forwards, backwards and attempt to use SEEK_END expecting EINVAL
- Partial read (alias reading past an allocated region)
- Read after exit with open FD
- Attempt to open `mem` from different process as different, unprivileged user expecting EPERM
- Open `mem` from different process as different, unprivileged user but with SYS_CAP_PTRACE
- Read `mem` from different process with same UID/GID